diff --git a/modules/mailchimp_signup/src/Form/MailchimpSignupForm.php b/modules/mailchimp_signup/src/Form/MailchimpSignupForm.php
index feaacfc..a6b11de 100644
--- a/modules/mailchimp_signup/src/Form/MailchimpSignupForm.php
+++ b/modules/mailchimp_signup/src/Form/MailchimpSignupForm.php
@@ -214,6 +214,32 @@ class MailchimpSignupForm extends EntityForm {
           '#disabled' => $mergevar->required,
         ];
       }
+
+        // TODO : - Multi audience tags
+        //        - Get Audience tag from Mailchimp or create a custom one
+        
+        $form['mc_lists_config']['audiencetags'] = [
+          '#prefix' => '<div id="audiencetags-wrapper">',
+          '#suffix' => '</div>',
+        ];
+    
+        $form['mc_lists_config']['audiencetags'] = [
+          '#type' => 'fieldset',
+          '#title' => $this->t('Audience Tag'),
+          '#description' => $this->t('Associate this form with a custom audience tag.'),
+          '#id' => 'mergefields-wrapper',
+          '#tree' => TRUE,
+          '#weight' => 20,
+        ];
+  
+        $form['mc_lists_config']['audiencetags']['audiencetag'] = [
+          '#type' => 'textfield',
+          '#title' => 'Audience Tag',
+          '#default_value' => isset($signup->settings['audiencetag']) ? $signup->settings['audiencetag'] : NULL,
+          '#required' => FALSE,
+          '#disabled' => FALSE,
+        ];
+
     }
 
     $form['subscription_settings'] = [
@@ -443,6 +469,7 @@ class MailchimpSignupForm extends EntityForm {
     }
 
     $signup->settings['mergefields'] = $mergefields;
+    $signup->settings['audiencetag'] = $form_state->getValue('audiencetags')['audiencetag'];
     $signup->settings['description'] = $form_state->getValue('description');
     $signup->settings['doublein'] = $form_state->getValue('doublein');
     $signup->settings['include_interest_groups'] = $form_state->getValue('include_interest_groups');
diff --git a/modules/mailchimp_signup/src/Form/MailchimpSignupPageForm.php b/modules/mailchimp_signup/src/Form/MailchimpSignupPageForm.php
index d96ad67..44bd62d 100644
--- a/modules/mailchimp_signup/src/Form/MailchimpSignupPageForm.php
+++ b/modules/mailchimp_signup/src/Form/MailchimpSignupPageForm.php
@@ -309,6 +309,8 @@ class MailchimpSignupPageForm extends FormBase {
     // Filter out blank fields so we don't erase values on the Mailchimp side.
     $mergevars = array_filter($form_state->getValue('mergevars'));
 
+    $audiencetag = array($this->signup->settings['audiencetag']);
+
     $email = $mergevars['EMAIL'];
 
     $gdpr_consent = $form_state->getValue('gdpr_consent');
@@ -350,7 +352,7 @@ class MailchimpSignupPageForm extends FormBase {
           $interests[] = $current_interests;
         }
       }
-      $result = mailchimp_subscribe($list_id, $email, $mergevars, $interests, $this->signup->settings['doublein'], 'html', NULL, $gdpr_consent);
+      $result = mailchimp_subscribe($list_id, $email, $mergevars, $interests, $this->signup->settings['doublein'], 'html', NULL, $gdpr_consent, $audiencetag);
 
       if (empty($result)) {
         $this->messenger->addWarning($this->t('There was a problem with your newsletter signup to %list.', [
